From 89fda5a9b587e2eaafdf7d63f8bcb1ec1e3ba698 Mon Sep 17 00:00:00 2001
From: Steve Loughran <stevel@apache.org>
Date: Thu, 12 May 2016 18:59:56 +0100
Subject: [PATCH 2195/2848] CDH-49898: MAPREDUCE-6639 Process hangs in
 LocatedFileStatusFetcher if FileSystem.get
 throws. Ryan Blue via stevel

(cherry picked from commit 7eddecd357014d4793df4bf2e5d987add02289f5)

Change-Id: I1046bcb2b1d7d8c2d05d77d0e0661515209ca1b5
---
 .../hadoop/mapred/LocatedFileStatusFetcher.java    |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/LocatedFileStatusFetcher.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/LocatedFileStatusFetcher.java
index 87114ad..3c8732a 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/LocatedFileStatusFetcher.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/LocatedFileStatusFetcher.java
@@ -175,7 +175,7 @@ private void registerInvalidInputError(List<IOException> errors) {
   private void registerError(Throwable t) {
     lock.lock();
     try {
-      if (unknownError != null) {
+      if (unknownError == null) {
         unknownError = t;
         condition.signal();
       }
@@ -368,4 +368,4 @@ public void onFailure(Throwable t) {
       registerError(t);
     }
   }
-}
\ No newline at end of file
+}
-- 
1.7.9.5

